.github/workflows/bindings_go.yml

```diff
@@ -0,0 +1,78 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Bindings Go CI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "core/**"
+      - "bindings/c/**"
+      - "bindings/go/**"
+      - ".github/workflows/bindings_go.yml"
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup
+
+      - name: Build c binding
+        working-directory: bindings/c
+        run: make build
+
+      - name: Check diff
+        run: git diff --exit-code
+
+      - name: Generate pkg-config file
+        run: |
+          echo "libdir=$(pwd)/target/debug/" >> opendal_c.pc
+          echo "includedir=$(pwd)/bindings/c/include/" >> opendal_c.pc
+          echo "Name: opendal_c" >> opendal_c.pc
+          echo "Description: opendal c binding" >> opendal_c.pc
+          echo "Version: 0.0.1" >> opendal_c.pc
+          echo "Libs: -L\${libdir} -lopendal_c" >> opendal_c.pc
+          echo "Cflags: -I\${includedir}" >> opendal_c.pc
+
+          echo "PKG_CONFIG_PATH=$(pwd)" >> $GITHUB_ENV
+
+          cat opendal_c.pc
+
+      - name: Run tests
+        working-directory: bindings/go
+        run: go test -tags dynamic .
```

Comment:
We need to correctly set the `LD_LIBRARY_PATH` in order for Go to find the libopendal.

Reply:
fixed it